cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRunner.java

```diff
@@ -245,18 +244,24 @@ public ProgramController call() throws Exception {
                                                                                  launchConfig.getLaunchOrder(),
                                                                                  localizeResources,
                                                                                  createEventHandler(cConf, options));
-
           TwillPreparer twillPreparer = twillRunner.prepare(twillApplication);
 
           // Add the configuration to container classpath
           twillPreparer.withResources(hConfFile.toURI(), cConfFile.toURI());
 
-          for (Map.Entry<String, RunnableResource> entry : launchConfig.getRunnables().entrySet()) {
+          // Setup log level
+          twillPreparer.setLogLevels(
+            transformLogLevels(SystemArguments.getLogLevels(options.getUserArguments().asMap())));
+
+          // Setup per runnable configurations
+          for (Map.Entry<String, RunnableDefinition> entry : launchConfig.getRunnables().entrySet()) {
             String runnable = entry.getKey();
-            RunnableResource runnableResource = entry.getValue();
-            if (runnableResource.getMaxRetries() != null) {
-              twillPreparer.withMaxRetries(runnable, runnableResource.getMaxRetries());
+            RunnableDefinition runnableDefinition = entry.getValue();
+            if (runnableDefinition.getMaxRetries() != null) {
+              twillPreparer.withMaxRetries(runnable, runnableDefinition.getMaxRetries());
             }
+            twillPreparer.setLogLevels(runnable, transformLogLevels(runnableDefinition.getLogLevels()));
+            twillPreparer.withConfiguration(runnable, runnableDefinition.getTwillConfigs());
```

Comment:
is it getTwillContainerConfigs()?

Reply:
More like `getTwillRunnableConfigs()`. Will update the name.